executor/insert_common.go

```diff
@@ -212,6 +214,11 @@ func insertRows(ctx context.Context, base insertCommon) (err error) {
 		evalRowFunc = e.evalRow
 	}
 
+	_, err = e.ctx.Txn(true)
+	if err != nil {
+		return err
+	}
```

Comment:
why need this? hard to understand.

Reply:
Start the transaction early to calculate shard values with `startTS`.

Reply:
Can we push this call as late as possible?

Reply:
+1, seems weird here.

Reply:
The transaction start time is needed in `evalRowFunc` [line 222](https://github.com/pingcap/tidb/pull/13127/files#diff-0f31e77a0899889b749ff90c69acaaf6R222).